Userland/Libraries/LibC/grp.cpp

```diff
@@ -53,6 +55,18 @@ struct group* getgrgid(gid_t gid)
     return nullptr;
 }
 
+int getgrgid_r(gid_t gid, struct group* group_buf, char* buffer, size_t buffer_size, struct group** group_entry_ptr)
+{
+    setgrent();
+    ScopeGuard guard = [] { endgrent(); };
```

Comment:
Since we're using getgrent_r we don't need (or want) to setgrent/endgrent in either of these new APIs.